hack/docgen/main.go

```diff
@@ -332,10 +334,6 @@ func collectFields(s *structType) (fields []*Field) {
 		yamlTag := tag.Get("yaml")
 		yamlTag = strings.Split(yamlTag, ",")[0]
 
-		if yamlTag == "" {
-			yamlTag = strings.ToLower(yamlTag)
```

Comment:
Not sure what was the point :)

Reply:
probably the intention was to do `strings.ToLower(name)`

Reply:
Added a check to require an explicit name